courses/models.py

```diff
@@ -567,3 +568,115 @@ class ProgramEnrollmentAudit(AuditModel):
     @classmethod
     def get_related_field_name(cls):
         return "enrollment"
+
+
+class CourseRunGrade(TimestampedModel, AuditableModel, ValidateOnSaveMixin):
+    """
+    Model to store course run final grades
+    """
+
+    user = models.ForeignKey(User, null=False, on_delete=models.CASCADE)
+    course_run = models.ForeignKey(CourseRun, null=False, on_delete=models.CASCADE)
+    grade = models.FloatField(
+        null=False, validators=[MinValueValidator(0.0), MaxValueValidator(1.0)]
+    )
+    letter_grade = models.CharField(max_length=1, null=True)
```

Comment:
wouldn't it be better to limit choices for this field ?

Reply:
It is a model to store course run final grades via edx_api client.

Reply:
I believe the data in edx for this is free text. In some courses it might be one of ["passed","failed"]. In other courses it might be one of ["A","B","C","D"].

Reply:
right, in that case we should not limit it to only one character via `max_length=1`.

Reply:
I am not sure about ["passed","failed"] but have thoughts for ["A","B","C","D"] at that point.